### PR TITLE
DroneCAN branding rename

### PR DESCRIPTION
**Pull request by coderkalyan**
WIP

Migrate UAVCAN documentation to DroneCAN + Cyphal. Takes the opportunity to clarify and expand on the existing docs.

* [x] Turned UAVCAN ESCs into isolated generic ESC product pages
* [x] New page for VESC ESCs
* [x] Updated product pages for peripherals
* [x] Updated docs navigation to match a new set of documents
* [x] Better explanation of getting started with DroneCAN
* [x] Firmware update documentation
* [x] Wiring updates

### CODE REVIEW

**Comment by hamishwillee** [comment]
Looks like a good start. Remember that you will need to also deal with (integrate any remaining info from) https://docs.px4.io/main/en/uavcan/#uavcan  and then remove it.

Also in development section we have https://docs.px4.io/main/en/uavcan/developer.html and its sub docs.

**Comment by coderkalyan** [comment]
@hamishwillee This is ready for your review. I plan to add more articles in the future, including information on the DroneCAN GUI, more developer information, and the control allocation stuff. However I think we should get these basic improvements in first.

FYI @mrpollo

File: en/SUMMARY.md [hunk 1]
```diff
@@ -291,8 +295,11 @@
     * [I2C Peripherals](sensor_bus/i2c_general.md)
       * [I2C bus accelerators](sensor_bus/i2c_general.md#i2c-bus-accelerators)
       * [TFI2CADT01 I2C address translator](sensor_bus/translator_tfi2cadt.md)
-    * [UAVCAN Peripherals](uavcan/README.md)
-      * [ARK CANnode](uavcan/ark_cannode.md)
+    * [CAN Peripherals](can/README.md)
+    * [DroneCAN Peripherals](dronecan/README.md)
+      * [DroneCAN Support Table](dronecan/support.md)
```
Comment (hamishwillee) [inline en/SUMMARY.md:300]: This file exists but is empty (zero bytes).

File: en/dronecan/ark_flow.md [hunk 2]
```diff
@@ -97,46 +101,6 @@ Parameter | Description
 --- | ---
 <a id="CANNODE_TERM"></a>[CANNODE_TERM](../advanced_config/parameter_reference.md#CANNODE_TERM) | CAN built-in bus termination.
 
-
-## Building Ark Flow Firmware
```
Comment (hamishwillee) [inline en/dronecan/ark_flow.md:101]: Ah, so this has been pushed into "PX4 DroneCAN Firmware". That works.

File: en/uavcan/bootloader_installation.md [hunk 3]
```diff
@@ -1,118 +0,0 @@
-# UAVCAN Bootloader Installation
```
Comment (hamishwillee) [inline en/uavcan/bootloader_installation.md:1]: Why is this deleted/where did the information go?

File: en/uavcan/developer.md [hunk 4]
```diff
@@ -1,29 +1,7 @@
...
-:::note
-[Hardware > UAVCAN Peripherals](../uavcan/README.md) contains information about using existing/supported UAVCAN components with PX4.
-:::
-
-## Upgrading Node Firmware
-
-The PX4 middleware will automatically upgrade firmware on UAVCAN nodes if the matching firmware is supplied.
-The process and requirements are described on the [UAVCAN Firmware](../uavcan/node_firmware.md) page.
-
 ### Debugging with Zubax Babel
```
Comment (hamishwillee) [inline en/uavcan/developer.md:3]: This is the only topic left in UAVCAN development. Is it/can it/should it go somewhere else? 

Or to put it another way, what happened to the "development story"

File: en/uavcan/node_firmware.md [hunk 5]
```diff
@@ -1,25 +1,3 @@
...
-To flash the UAVCAN device, all you need to do is copy the UAVCAN firmware binary into the root directory of the flight controller’s SD card and reboot.
-
-After restarting the flight controller the UAVCAN device will automatically be updated with new firmware.
-If successful the firmware binary will be removed from the root directory and there will be a file named **XX.bin** in the **/ufw** directory of the SD card.
-
-:::note
-UAVCAN firmware is usually created as part of the build process.
-Commonly the firmware image is built in **build/_name_of_your_uavcan_device_**,  where _name_of_your_uavcan_device_ is specific to the device you are flashing (e.g. **build/ark_can-flow_default**), and will be named with the format **XX-X.X.XXXXXXXX.uavcan.bin**.
-:::
-
```
Comment (hamishwillee) [inline en/uavcan/node_firmware.md:22]: This doc seems to have been partially stripped. What is the plan here?

The idea would be to remove all mentions of UAVCAN from here. Either make this stuff cyphal or dronecan specific, or push it into those docs sections.

File: en/uavcan/README.md [hunk 6]
```diff
@@ -1,6 +1,10 @@
+<Redirect to="../can/" />
```
Comment (hamishwillee) [inline en/uavcan/README.md:1]: OK, I added this to redirect the old uavcan doc to can. Was that your ultimate plan?

What I'm trying to work out is what you intend for this doc? I assumed at the end of this process that all uavcan docs would be deleted. So my concern is that this still has information that should be kept. For example the PX4 configuraiton stuff like " `0`: UAVCAN driver disabled." (see below).

File: en/dronecan/README.md [hunk 7]
```diff
@@ -0,0 +1,68 @@
+# DroneCAN
+
+From the DroneCAN website:
+
+> DroneCAN is the primary CAN protocol used by the ArduPilot and PX4 projects for communication with CAN peripherals.
+> It is an open protocol with open communication, specification and multiple open implementations.
```
Comment (hamishwillee) [inline en/dronecan/README.md:6]: Let's just link to the website. I'll capture the essentials

File: en/can/README.md [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+PX4 supports two software protocols for communicating with CAN devices:
+
+- [DroneCAN](../dronecan/README.md): PX4 recommends this for most common setups.
+  It is well supported by PX4, is a mature product with extensive peripheral support, and has had years of testing.
+- [Cyphal](https://opencyphal.org): PX4 support is a "work in progress".
+  Cyphal is a much newer protocol which allows more flexibility and configuration, especially on larger and more complex vehicles.
+  It has not yet seen significant adoption.
+
+:::note
+Both DroneCAN and Cyphal originate from an earlier project named UAVCAN.
```
Comment (hamishwillee) [inline en/can/README.md:16]: Note, I'm slightly hiding the community disagreement here. It's obvious something happened, but irrelevant other than for people to know the mapping

File: en/can/README.md [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+
+DroneCAN and Cyphal support using a second (redundant) CAN interface.
+This is completely optional but increases the robustness of the connection.
+All Pixhawk flight controllers come with 2 CAN interfaces; if your peripherals support 2 CAN interfaces as well, it is recommended to wire both up for increased safety.
+
+## Firmware
+
+Some CAN peripherals come with custom firmware written by the manufacturer.
+However, in addition to firmware for flight controllers, PX4 can be run as peripheral firmware itself on supported devices.
+See the [DroneCAN Support Table](../dronecan/support.md) for more information.
```
Comment (hamishwillee) [inline en/can/README.md:61]: This is the wrong link. I think you want the firmware page.

File: en/can/README.md [hunk 10]
```diff
@@ -0,0 +1,88 @@
...
+
+## Support and Configuration
+
+[DroneCAN Setup and Configuration](../dronecan/README.md)
+
+[DroneCAN Support Table](../dronecan/support.md)
```
Comment (hamishwillee) [inline en/can/README.md:67]: This is empty. It is supposed to be the set of supported devices right? I'd be happy for that to be in the DroneCan docs OR it could be in its own docs. If it is in its own doc as implied, then support is a bad name because it implies "how do I get support for DroneCAN problems" not "what devices are supported"

So this would be DroneCan Devices or similar.

It should be something like

**Comment by hamishwillee** [comment]
Thanks @coderkalyan .
What I think you have done is pretty much:

- moved uavcan docs to /dronecan
- added a /can/README to introduce both protocols
- Split out some of the docs that reproduced info - like the PX4 firmware section.
- Split out some of the developer stuff as well.
 
Looks like a very good start. 
- There are some loose bits that have not yet migrated into the right place - e.g. remaining uavcan/README, the supported device list, a few parts of the UAVCAN developer.
- When you're finished all that there are more references to UAVCAN in the docs - e.g. http://docs.px4.io/main/en/assembly/cable_wiring.html#uavcan-cables . A search over the whole doc source should follow getting the core of this right.
- Cyphal is only referenced on the CAN page. That's not enough info to know how to enable it on PX4. Is that because there aren't any peripherals yet?

File: en/dronecan/ark_flow.md (hunk 2 above)
Comment (coderkalyan) [reply en/dronecan/ark_flow.md:101]: Yes. I thought we should keep things in a central place instead of repeating information.

File: en/uavcan/bootloader_installation.md (hunk 3 above)
Comment (coderkalyan) [reply en/uavcan/bootloader_installation.md:1]: I decided listing multiple ways to flash bootloaders via SWD is out of scope for the PX4 docs, there's tons of materials on the web on how to do it for various devices, better than we could write. However, see cannode_fw.md#dronecan-bootloader-information, where I wrote a basic summary.

File: en/uavcan/developer.md (hunk 4 above)
Comment (coderkalyan) [reply en/uavcan/developer.md:3]: Ah, thanks for catching this, I didn't take a close enough look at the diff. The reason I'm not migrating the Zubax Babel information right now is because I'd like to revisit it in 2-4 weeks pending some research/testing on the state of the dronecan gui tool and also implementing a mavlink slcan tunnel into PX4. I'll be pulling this out into a full article on CAN development.

If you'd prefer, I can write a stub for that article now with this in it.

Comment (hamishwillee) [reply en/uavcan/developer.md:3]: @coderkalyan Thanks. 

Yes

1. The documentation should always be complete and ready to publish. So yes, we need to clean this.
2. We shouldn't lose information "accidentally"
3. All links to old docs still end up at sensible end points.

If you can create a reasonable stub article then:
- Create new doc for DroneCan or Can development that includes this section, and any other links and information you can find quickly. Perhaps a note that this section is being reconstructed. I would be happy to have links to firmware updates etc as well. 
- Redirect the old doc to the new doc.
- Fix up links to the new docs  (even with the redirects, this is better)

File: en/uavcan/node_firmware.md (hunk 5 above)
Comment (hamishwillee) [reply en/uavcan/node_firmware.md:22]: @coderkalyan Same story for this. Some of it moved out, but is the story on this page still consistent (a question, not a statement)

File: en/uavcan/README.md [hunk 11]
```diff
@@ -71,6 +75,7 @@ In order to use UAVCAN components with PX4 you will first need to enable the UAV
 1. [UAVCAN_ENABLE](../advanced_config/parameter_reference.md#UAVCAN_ENABLE) must be [set](../advanced_config/parameters.md) to one of the non-zero values.
```
Comment (hamishwillee) [inline en/uavcan/README.md:75]: @coderkalyan In particular this info is the PX4 configuration. It gets referenced everywhere and must be visible after the update.

File: en/uavcan/bootloader_installation.md (hunk 3 above)
Comment (hamishwillee) [reply en/uavcan/bootloader_installation.md:1]: @coderkalyan I agree we shouldn't list the ways, but I do think that providing links to the instructions would be useful. In "PX4 DroneCAN Firmware" you link to the main product guides for the different ways of doing this - but if there is a more specific link for installing it would be good to use that.

Further, this page was a generic bootloader page, so this is where you'd come for instructions on any firmware. I am not sure someone would expect to find the information for random CAN firmware in "PX4 DroneCAN Firmware".

In other words, this is a question of discoverability. 

"Somewhere" we should say that the installation is covered by the manufacturer for the different probes, and that for dronecan you can find the information at Xxxx. That might be in the top level CAN page?

**Comment by hamishwillee** [comment]
I will review again when you've done another round of iteration and provided a response for all the comments.

File: en/uavcan/README.md (hunk 6 above)
Comment (coderkalyan) [reply en/uavcan/README.md:1]: I moved the rest of the stuff out.

File: en/can/README.md (hunk 9 above)
Comment (coderkalyan) [reply en/can/README.md:61]: Yeah, you're right. Thanks, just fixed.

File: en/advanced_config/esc_calibration.md [hunk 12]
```diff
@@ -1,7 +1,7 @@
 # ESC Calibration
 
 :::note
-These instructions are only relevant to [PWM ESCs](../peripherals/pwm_escs_and_servo.md) and [OneShot ESCs](../peripherals/oneshot.md) ([DShot](../peripherals/dshot.md) and [UAVCAN](../uavcan/escs.md) ESCs do not require calibration).
+These instructions are only relevant to [PWM ESCs](../peripherals/pwm_escs_and_servo.md) and [OneShot ESCs](../peripherals/oneshot.md) ([DShot](../peripherals/dshot.md) and [DroneCAN](../dronecan/escs.md) ESCs do not require calibration).
```
Comment (hamishwillee) [inline en/advanced_config/esc_calibration.md:4]: @coderkalyan Isn't this "CAN" ESCS. I think maybe we can still link to the dronecan page though:

```suggestion
These instructions are only relevant to [PWM ESCs](../peripherals/pwm_escs_and_servo.md) and [OneShot ESCs](../peripherals/oneshot.md) ([DShot](../peripherals/dshot.md) and [CAN](../dronecan/escs.md) ESCs do not require calibration).

```

File: en/dronecan/README.md [hunk 13]
```diff
@@ -0,0 +1,114 @@
...
+  - [ARK Flow](ark_flow.md)
+  - [Avionics Anonymous Laser Altimeter UAVCAN Interface](../dronecan/avanon_laser_interface.md)
+- Optical Flow
+  - [Ark Flow](ark_flow.md)
+- Generic CAN Node (enables use of I2C, SPI, UART sensors on the CAN bus).
+  - [ARK CANnode](../dronecan/ark_cannode.md)
+
+## Hardware Setup
+
+DroneCAN operates over CAN. Refer to the [CAN setup documentation](../can/README.md) before continuing with this article.
```
Comment (hamishwillee) [inline en/dronecan/README.md:50]: Probably best to make the link go to the specific wiring section. Also updating text to my taste.

```suggestion
DroneCAN operates over a CAN network.
DroneCAN hardware should be connected as described in [CAN > Wiring](../can/README.md#wiring).
```

File: en/uavcan/README.md (hunk 6 above)
Comment (hamishwillee) [reply en/uavcan/README.md:1]: Excellent!!!!

File: en/can/README.md [hunk 14]
```diff
@@ -0,0 +1,88 @@
...
+
+DroneCAN and Cyphal support using a second (redundant) CAN interface.
+This is completely optional but increases the robustness of the connection.
+All Pixhawk flight controllers come with 2 CAN interfaces; if your peripherals support 2 CAN interfaces as well, it is recommended to wire both up for increased safety.
+
+## Firmware
+
+Some CAN peripherals come with custom firmware written by the manufacturer.
+However, in addition to firmware for flight controllers, PX4 can be run as peripheral firmware itself on supported devices.
+See [PX4 DroneCAN Firmware](../dronecan/px4_cannode_fw.md.md) for more information.
```
Comment (hamishwillee) [inline en/can/README.md:61]: Invalid link format

```suggestion
See [PX4 DroneCAN Firmware](../dronecan/px4_cannode_fw.md) for more information.
```

File: en/dronecan/cuav_can_pmu.md [hunk 15]
```diff
@@ -1,13 +1,18 @@
...
-CAN PMU<sup>&reg;</sup> is a high-precision [UAVCAN](../uavcan/README.md) power module developed by CUAV<sup>&reg;</sup>.
+CAN PMU<sup>&reg;</sup> is a high-precision [DroneCAN](README.md) power module developed by CUAV<sup>&reg;</sup>.
 It runs the CUAV ITT compensation algorithm, which enables drones to get the battery data more accurately.
 
 ![CAN PMU](../../assets/hardware/power_module/cuav_can/can_pmu.jpg)
 
 It is recommended for use in large commercial vehicles, but might also be used for research vehicles.
 
-## Specifications
+## Where to Buy
```
Comment (hamishwillee) [inline en/dronecan/cuav_can_pmu.md:10]: Hmmm. I don't have a problem with Where to Buy, but we are now inconsistent. I might use this PR to fix them all up.

File: en/peripherals/vesc.md [hunk 16]
```diff
@@ -0,0 +1,99 @@
...
+  :::
+- [Zubax Mitochondrik](https://zubax.com/products/mitochondrik) - integrated sensorless PMSM/BLDC motor controller chip (used in ESCs and integrated drives)
+  - [Zubax Sadulli Integrated Drive](https://shop.zubax.com/collections/integrated-drives/products/sadulli-integrated-drive-open-hardware-reference-design-for-mitochondrik?variant=27740841181283)
+- [OlliW’s UC4H ESC-Actuator Node](http://www.olliw.eu/2017/uavcan-for-hobbyists/#chapterescactuator)
+- A number of others are [listed here](https://forum.uavcan.org/t/uavcan-esc-options/452/3?u=pavel.kirienko)
+
+:::note
+This list is *not exhaustive/complete*.
+If you know of another ESC, please add it to the list!
+:::
```
Comment (hamishwillee) [inline en/peripherals/vesc.md:43]: @coderkalyan Does this info actually apply to VESC? This looks like a copy paste error - or at least this is duplication of the higher level docs. Can you check this doc?

File: en/dronecan/zubax_orel.md [hunk 17]
```diff
@@ -0,0 +1,20 @@
...
+The Zubax Orel 20 is an CAN ESC designed to run the open source [PX4 Sapog ESC Firmware](dronecan/sapog.md).
+
+While it can be controlled using traditional PWM input, it is designed to operate over CAN bus using [DroneCAN](README.md).
+This has multiple benefits:
+
+- CAN has been specifically designed to deliver robust and reliable connectivity over relatively large distances.
+  It enables safe use of ESCs on bigger vehicles and communication redundancy.
+- The bus is bi-directional, enabling health monitoring, diagnostics, and RPM telemetry.
+- Wiring is less complicated as you can have a single bus for connecting all your ESCs and other DroneCAN peripherals.
+- Setup is easier as you configure ESC numbering by manually spinning each motor.
```
Comment (hamishwillee) [inline en/dronecan/zubax_orel.md:12]: @coderkalyan This text is reproduced in quite a few of these docs. I don't think there is real value in repeating it. I will remove.

File: en/dronecan/README.md [hunk 18]
```diff
@@ -0,0 +1,139 @@
...
+PX4 requires an SD card to enable dynamic node allocation and for firmware update.
+The SD card is not used in flight.
+:::
+
+:::note
+DroneCAN was previously known as UAVCAN v0 (or just UAVCAN).
+The name was changed in 2022.
+:::
+
+## Benefits of DroneCAN
```
Comment (hamishwillee) [inline en/dronecan/README.md:15]: @coderkalyan FYI only, I pulled the "benefits" out of the heading above and made them a section with its own list. I wasn't entirely sure of this, but it does make things easier to read, and means that if we want to say in other places "CAN has many benefits" we have a good place to link.

**Comment by hamishwillee** [comment]
@coderkalyan Thanks for all your work on this. I _think_ its looking pretty good. The only outstanding question is around whether the VESC doc contains some stuff that isn't needed - as per https://github.com/PX4/PX4-user_guide/pull/1994/files#r947400736


I've subedited. I also deleted the old uavcan logo and the support page you added - since that info is in the parent doc. 

Let me know when you've had a look at the vesc page again and we can then get it in.

File: en/peripherals/vesc.md (hunk 16 above)
Comment (coderkalyan) [reply en/peripherals/vesc.md]: Yes, you're right. Copy paste error :) Fixed.

File: en/dronecan/README.md (hunk 18 above)
Comment (coderkalyan) [reply en/dronecan/README.md:15]: Sure, sounds good.

**Comment by coderkalyan** [comment]
@hamishwillee is this ready to get in? Can you please take another look and let me know if there's anything else you want added?

I fixed the vesc page and I think everything else has been addressed.

**Comment by mcsauder** [comment]
Hi Everyone here,

I'm new to this PR but recently acquainted with how faceted the "CAN" aspects of PX4 have become in the past few years.  There seem to be at least a few different companies servicing CAN objects, but nothing is coalescing into a coherent standard.

Libuavcan, pyavcan, DSDL, etc., is currently is a breaking feature in PX4 with current Ubuntu LTS. Cyphal is recently rebranded by Zubax, DroneCan is mostly Ardupilot and isn't being managed within PX4 as far as I can tell, and CANNode is a new thing that needs additional accomodation.

At present, it is a big mess.  Is someone willing to form a working committee to iron this stuff out that the automotive industry already has? :}

Only my 2 cents, but currently also a pain in my backside only because I can't compile code in Ubuntu 22.04 because of this mess.  How are we going to clean things up?

I have no answers... :)

Thank you for your work @hamishwillee and @coderkalyan !

-Mark

File: en/peripherals/vesc.md [hunk 19]
```diff
@@ -0,0 +1,51 @@
...
+
+(Optional) Set [UAVCAN_ESC_IDLT](../advanced_config/parameter_reference.md#UAVCAN_ESC_IDLT) to 1 in order to ensure that the motors are always running at least at the idle throttle while the system is armed.
+
+:::note
+Some systems will not benefit from this behavior, e.g. glider drones).
+:::
+
+## Troubleshooting
+
+See [DroneCAN Troubleshooting](README.md#troubleshooting)
```
Comment (hamishwillee) [inline en/peripherals/vesc.md:47]:
```suggestion
See [DroneCAN Troubleshooting](README.md#troubleshooting).
```

File: en/peripherals/vesc.md [hunk 20]
```diff
@@ -0,0 +1,51 @@
...
+However the `UAVCAN ESC Index` starts from `0`, and as such it is always one index lower than the `VESC ID`.
+For example, in a quadcopter the bottom left motor will have `VESC ID = 2` and `UAVCAN ESC Index = 1`.
+
+Finally the `CAN Baud Rate` must match the value set in [UAVCAN_BITRATE](../advanced_config/parameter_reference.md#UAVCAN_BITRATE).
+
+## Flight Controller Setup
+
+### Enable DroneCAN
+
+Connect the ESCs to the Pixhawk CAN bus. Power up the entire vehicle using a battery or power supply (not just the flight controller over USB) and enable the DroneCAN driver by setting the parameter [UAVCAN_ENABLE](../advanced_config/parameter_reference.md#UAVCAN_ENABLE) to '3' to enable both dynamic node ID allocation and DroneCAN ESC output.
```
Comment (hamishwillee) [inline en/peripherals/vesc.md:35]:
```suggestion
Connect the ESCs to the Pixhawk CAN bus. Power up the entire vehicle using a battery or power supply (not just the flight controller over USB) and enable the DroneCAN driver by setting the parameter [UAVCAN_ENABLE](../advanced_config/parameter_reference.md#UAVCAN_ENABLE) to `3` to enable both dynamic node ID allocation and DroneCAN ESC output.
```

File: en/peripherals/vesc.md [hunk 21]
```diff
@@ -0,0 +1,51 @@
...
+While it can be controlled using traditional PWM input, it also supports being connected over CAN bus using [DroneCAN](../dronecan/README.md).
+
+## Hardware Setup
+### Wiring
+
+ESCs are connected to the CAN bus using the VESC CAN connector. Note that this is *not* the Pixhawk standard 4 pin JST GH connector. For more information, refer to the [CAN Wiring](../can/README.md#wiring) instructions. ESC order does not matter.
+
+## Firmware Setup
+
+The preferred tool for motor enumeration is the [VESC tool](https://vesc-project.com/vesc_tool). In addition to the normal motor configuration that you will have to setup in the VESC tool, you will also need to properly setup the app configuration.
```
Comment (hamishwillee) [inline en/peripherals/vesc.md:13]:
```suggestion
The preferred tool for motor enumeration is the [VESC tool](https://vesc-project.com/vesc_tool).
In addition to the normal motor configuration that you will have to setup in the VESC tool, you will also need to properly setup the app configuration.
```

File: en/peripherals/vesc.md [hunk 22]
```diff
@@ -0,0 +1,51 @@
+# VESC ESCs (DroneCAN)
+
+The VESC project is a fully open source hardware and software design for advanced FOC motor controllers.
```
Comment (hamishwillee) [inline en/peripherals/vesc.md:3]:
```suggestion
The [VESC project](https://vesc-project.com/) is a fully open source hardware and software design for advanced FOC motor controllers.
```

File: en/peripherals/vesc.md [hunk 23]
```diff
@@ -0,0 +1,51 @@
...
+While it can be controlled using traditional PWM input, it also supports being connected over CAN bus using [DroneCAN](../dronecan/README.md).
+
```
Comment (hamishwillee) [inline en/peripherals/vesc.md:5]:
```suggestion

## Where to Buy

[Vesc Project > Hardware](https://vesc-project.com/Hardware)

```

**Comment by hamishwillee** [comment]
@coderkalyan Looks good to me! Thanks so much for your hard work on this. As I understand it you will be coming back to this to look at the developer story a bit more in a month or so right?

**Comment by hamishwillee** [comment]
@mcsauder 

Re https://github.com/PX4/PX4-user_guide/pull/1994#issuecomment-1223571595

Thanks. Unfortunately "everyone here" might be just you, me and @coderkalyan. 

W.r.t. fragmentation, AFAIK DroneCAN is just UAVCAN 0 with some new and better designed messages. ArduPilot started to  split out because they didn't like where UAVCAN 1.0 was going. It's pretty much the same old DroneCAN though but with some evolution of the messages. PX4 adopted this, and I understand @dagar has been engaged pretty closely to make sure we're all aligned.

Cyphal is UAVCAN 1.0 rebranded. ArduPilot do not appear interested. PX4 appear to be playing "wait and see" to determine the traction. 

Can't comment on the rest, but Daniel will know. 

What I would do is post this on the can channel on slack. For the build issues, I would create an actual issue. That won't resolve unless it gets looked at.

And yes, this kind of thing is frustrating.
